OsmAnd/src/net/osmand/plus/plugins/srtm/Relief3DFragment.java

```diff
@@ -0,0 +1,352 @@
+		stateTv = root.findViewById(R.id.state_tv);
+		iconIv = root.findViewById(R.id.icon_iv);
+		downloadContainer = root.findViewById(R.id.download_container);
+		downloadTopDivider = root.findViewById(R.id.download_container_top_divider);
+		downloadBottomDivider = root.findViewById(R.id.download_container_bottom_divider);
+		observableListView = root.findViewById(R.id.list_view);
+
+		TextView titleTv = root.findViewById(R.id.title_tv);
+		titleTv.setText(R.string.relief_3d);
+
+		switchCompat.setChecked(relief3DEnabled);
+		switchCompat.setOnClickListener(this);
+		UiUtilities.setupCompoundButton(switchCompat, nightMode, UiUtilities.CompoundButtonType.PROFILE_DEPENDENT);
+
+		setupContentCard(root);
+		updateUiMode();
+		return root;
+	}
+
+	public float getElevationScaleFactor() {
+		return srtmPlugin.getVerticalExaggerationScale();
+	}
+
+	private void setupContentCard(@NonNull View root) {
+		downloadDescriptionTv.setText(R.string.relief_3d_download_description);
+		View verticalExaggerationBtn = root.findViewById(R.id.vertical_exaggeration_button);
+		verticalExaggerationBtn.setOnClickListener(view -> {
+			MapActivity mapActivity = getMapActivity();
+			if (mapActivity != null) {
+				mapActivity.getDashboard().hideDashboard();
+				VerticalExaggerationFragment.showInstance(mapActivity.getSupportFragmentManager());
+			}
+		});
+	}
+
+	private void showHideTopShadow(@NonNull View view) {
+		boolean portrait = AndroidUiHelper.isOrientationPortrait(requireActivity());
+		AndroidUiHelper.updateVisibility(view.findViewById(R.id.shadow_on_map), portrait);
+	}
+
+	@Override
+	public void onClick(View view) {
+		int id = view.getId();
+		if (id == R.id.switch_compat) {
+			onSwitchClick();
+		}
+	}
+
+	private void updateUiMode() {
+		if (relief3DEnabled) {
+			iconIv.setImageDrawable(uiUtilities.getPaintedIcon(R.drawable.ic_action_3d_relief, profileColor));
+			stateTv.setText(R.string.shared_string_on);
+			updateDownloadSection();
+		} else {
+			iconIv.setImageDrawable(uiUtilities.getIcon(
+					R.drawable.ic_action_3d_relief,
+					nightMode
+							? R.color.icon_color_secondary_dark
+							: R.color.icon_color_secondary_light));
+			stateTv.setText(R.string.shared_string_off);
+		}
+		exaggerationValueTv.setText(getFormattedScaleValue(app, getElevationScaleFactor()));
+		adjustGlobalVisibility();
+	}
+
+	private void adjustGlobalVisibility() {
+		titleBottomDivider.setVisibility(relief3DEnabled ? View.GONE : View.VISIBLE);
+		contentContainer.setVisibility(relief3DEnabled ? View.VISIBLE : View.GONE);
+	}
+
+	private void onSwitchClick() {
+		relief3DEnabled = !relief3DEnabled;
+		switchCompat.setChecked(relief3DEnabled);
+		settings.ENABLE_3D_MAPS.set(relief3DEnabled);
+		MapActivity mapActivity = getMapActivity();
+		if (mapActivity != null) {
+			app.runInUIThread(() -> app.getOsmandMap().getMapLayers().getMapInfoLayer().recreateAllControls(mapActivity));
+		}
+
+		updateUiMode();
+	}
+
+	private void updateDownloadSection() {
```

Comment:
Copy-paste from TerrainFragment, try to move similar code to one place (parent class?)